mycore-user2/src/main/java/org/mycore/user2/MCRUserAttributeMapper.java

```diff
@@ -113,88 +113,106 @@ public boolean mapAttributes(final Object object, final Map<String, ?> attribute
-                            if (attribute.converter != null) {
-                                convCls = (Class<? extends MCRUserAttributeConverter>) Class
-                                    .forName(attribute.converter);
-                            } else if (aConv != null) {
-                                convCls = aConv.value();
-                            }
-
-                            if (convCls != null) {
-                                MCRUserAttributeConverter converter = convCls.getDeclaredConstructor().newInstance();
-                                LOGGER.debug("convert value \"{}\" with \"{}\"", value, converter.getClass().getName());
-                                value = converter.convert(value,
-                                    attribute.separator != null ? attribute.separator : attrAnno.separator(),
-                                    attribute.getValueMap());
-                            }
-
-                            if (value != null || ((attrAnno.nullable() || attribute.nullable) && value == null)) {
-                                Object oldValue = getValue(object, annotated);
-                                if (oldValue != null && oldValue.equals(value)) {
-                                    continue;
-                                }
-
-                                if (annotated instanceof Field field) {
-
-                                    LOGGER.debug("map attribute \"{}\" with value \"{}\" to field \"{}\"",
-                                        attribute.mapping, value, field.getName());
-
-                                    field.setAccessible(true);
-                                    field.set(object, value);
-
-                                    changed = true;
-                                } else if (annotated instanceof Method method) {
-
-                                    LOGGER.debug("map attribute \"{}\" with value \"{}\" to method \"{}\"",
-                                        attribute.mapping, value, method.getName());
-
-                                    method.setAccessible(true);
-                                    method.invoke(object, value);
-
-                                    changed = true;
-                                }
-                            } else {
-                                throw new IllegalArgumentException(
-                                    "A not nullable attribute \"" + name + "\" was null.");
-                            }
-                        }
-                    }
+            for (Attribute attribute : attribs) {
+                if (!attributes.containsKey(attribute.mapping)) {
+                    continue;
+                }
+
+                Object value = attributes.get(attribute.mapping);
+                if (value == null) {
+                    LOGGER.warn("Could not apply mapping for {}", attribute.mapping);
+                    continue;
```

Comment:
The Code before would not continue, instead it would use the MCRUserAttributeConverter with an null value and abort later.
Since all the MCRUserAttributeConverter in mycore check for a null value, the behavior is probably the same.